### PR TITLE
Merge baremetal caph updates into ecf branch

### PR DESCRIPTION
**Pull request by romoh**
Move commits from BareMetal2 into ecf branch

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -23,4 +23,8 @@ require (
 	sigs.k8s.io/controller-runtime v0.5.2
 )
 
-replace github.com/Azure/go-autorest => github.com/Azure/go-autorest v13.4.0+incompatible
+replace (
```
Comment (romoh) [inline go.mod:26]: @jiria - what should we do here?

Comment (jiria) [reply go.mod:26]: We probably do not want to keep the overrides, right? Sorry for not noticing this earlier.

Comment (romoh) [reply go.mod:26]: Good catch, will send another PR.

Comment (romoh) [reply go.mod:26]: This was resolved before merging the PR
https://github.com/microsoft/cluster-api-provider-azurestackhci/pull/117/files

Comment (jiria) [reply go.mod:26]: Ah, thanks.

**Pull request merged**